### PR TITLE
Add datastorage disk

### PR DESCRIPTION
**Pull request by lbnjma**
Add datadisk to instances

### CODE REVIEW

**Comment by dtzar** [comment]
Looks good. I just don't want to force people to use a datastorage disk, so can you make it a conditional?

**Comment by lbnjma** [comment]
To make datadisk conditional I had to create two new resources named `vm-linux-with-datadisk` and `vm-windows-with-datadisk` and change the condition to use it. 
It may not the best way to do that, but due to HCL limitation I have not found any other solution.

File: main.tf [hunk 1]
```diff
@@ -75,8 +83,98 @@ resource "azurerm_virtual_machine" "vm-linux" {
...
+  storage_image_reference {
+    id        = "${var.vm_os_id}"
+    publisher = "${coalesce(var.vm_os_publisher, module.os.calculated_value_os_publisher)}"
+    offer     = "${coalesce(var.vm_os_offer, module.os.calculated_value_os_offer)}"
+    sku       = "${coalesce(var.vm_os_sku, module.os.calculated_value_os_sku)}"
+    version   = "${var.vm_os_version}"
+  }
+
+  storage_os_disk {
+    name              = "osdisk${count.index}"
```
Comment (dtzar) [inline main.tf]: Please change to `"osdisk-${var.vm_hostname}-${count.index}"` to make things consistent and prevent same named disk conflicts.

Please fix on line 195 as well.

File: main.tf [hunk 2]
```diff
@@ -48,11 +48,19 @@ resource "azurerm_virtual_machine" "vm-linux" {
   }
 
   storage_os_disk {
-    name          = "osdisk-${var.vm_hostname}-${count.index}"
+    name          = "osdisk${count.index}"
```
Comment (dtzar) [inline main.tf]: Please change to `"osdisk-${var.vm_hostname}-${count.index}"` to make things consistent and prevent same named disk conflicts.

File: variables.tf [hunk 3]
```diff
@@ -108,6 +108,22 @@ variable "delete_os_disk_on_termination" {
...
+  description = "Data Disk Storage Account type"
+  default = "Standard_LRS"
+}
+
+variable "data_disk_size_gb" {
+  description = "Storage data disk size size"
+  default = ""
+}
+
+variable "datadisk" {
```
Comment (dtzar) [inline variables.tf]: nit: keep names consistent with `_` so `data_disk`

File: main.tf [hunk 4]
```diff
@@ -75,8 +83,98 @@ resource "azurerm_virtual_machine" "vm-linux" {
...
+
+  storage_os_disk {
+    name              = "osdisk${count.index}"
+    create_option     = "FromImage"
+    caching           = "ReadWrite"
+    managed_disk_type = "${var.storage_account_type}"
+  }
+
+  storage_data_disk {
+    name = "datadisk-${var.vm_hostname}-${count.index}"
```
Comment (dtzar) [inline main.tf]: nit: keep spacing with `=` aligned throughout (i.e. here, lines 201-208, etc)

File: main.tf [hunk 5]
```diff
@@ -30,7 +30,7 @@ resource "azurerm_storage_account" "vm-sa" {
 }
 
 resource "azurerm_virtual_machine" "vm-linux" {
-  count = "${contains(list("${var.vm_os_simple}","${var.vm_os_offer}"), "WindowsServer") ? 0 : var.nb_instances}"
+  count = "${!contains(list("${var.vm_os_simple}","${var.vm_os_offer}"), "WindowsServer") && var.datadisk == "false" ? var.nb_instances : 0}"
```
Comment (dtzar) [inline main.tf]: This logic does not work for me.  I am unable to provision a VM without a datadisk.

**Comment by dtzar** [comment]
I updated the readme to be current with the code up there in this PR https://github.com/Azure/terraform-azurerm-compute/pull/17.  If you want to add the data disk option for the Windows VM in the "advanced" example, that'd be cool.

File: main.tf (hunk 5 above)
Comment (lbnjma) [reply main.tf]: Fixed. Regression from a bad merge :(

File: main.tf (hunk 2 above)
Comment (lbnjma) [reply main.tf]: Fixed.

File: main.tf (hunk 1 above)
Comment (lbnjma) [reply main.tf]: Fixed.

File: main.tf (hunk 4 above)
Comment (lbnjma) [reply main.tf]: Fixed.

File: variables.tf (hunk 3 above)
Comment (lbnjma) [reply variables.tf]: Fixed.

**Comment by dtzar** [comment]
LGTM. Thanks for your contribution!

**Pull request merged**